src/gamma_inc.jl

```diff
@@ -142,7 +141,250 @@ function rgammax(a::Float64,x::Float64)
             return (a*exp(t))*(1.0 + rgamma1pm1(a))
         end
     end
+end
+"""
+    auxgam(x)
+
+Compute function `g` in ``1/\\Gamma(x+1) = 1+x*(x-1)*g(x)``, -1 <= x <= 1 
+"""
+function auxgam(x::Float64)
+    dr = zeros(18)
+    if x < 0
+        return -(1.0+(1+x)*(1+x)*auxgam(1+x))/(1.0-x)
+    else
+        dr[1] = -1.013609258009865776949
+        dr[2] = 0.784903531024782283535e-1
+        dr[3] = 0.67588668743258315530e-2
+        dr[4] = -0.12790434869623468120e-2
+        dr[5] = 0.462939838642739585e-4
+        dr[6] = 0.43381681744740352e-5
+        dr[7] = -0.5326872422618006e-6
+        dr[8] = 0.172233457410539e-7
+        dr[9] = 0.8300542107118e-9
+        dr[10] = -0.10553994239968e-9
+        dr[11] = 0.39415842851e-11
+        dr[12] = 0.362068537e-13
+        dr[13] = -0.107440229e-13
+        dr[14] = 0.5000413e-15
+        dr[15] = -0.62452e-17
+        dr[16] = -0.5185e-18
+        dr[17] = 0.347e-19
+        dr[18] = -0.9e-21
+        t = 2*x - 1.0
+        return chepolsum(t,dr)
+    end
 end   
+"""
+    loggamma1p(a)
+
+Compute ``log(\\Gamma(1+a))`` for -1 < a <= 1.
+"""
+function loggamma1p(x::Float64)
+    return -log1p(x*(x-1.0)*auxgam(x))
+end 
+
+"""
+    chepolsum(n,x,a)
+
+Computes a series of Chebyshev Polynomials given by : a[0]/2 + a[1]T1(x) + .... + a[n]Tn(X)
+"""
+function chepolsum(x::Float64, a::Array{Float64,1})
+    n = length(a)
+    if n == 1
+        return a[1]/2.0
+    elseif n == 2
+        return a[1]/2.0 + a[2]*x
+    else
+        tx = 2*x
+        r = a[n]
+        h = a[n-1] + r*tx
+        for k = n-2:-1:2
+            s=r
+            r=h
+            h=a[k]+r*tx-s
+        end
+        return a[1]/2.0 - r + h*x
+    end
+end
+"""
+    stirling(x)
+
+Compute log(gamma(x)) - (x-0.5)*log(x) + x - log(2pi)/2 using stirling series for asymptotic part 
+in https://dlmf.nist.gov/5.11.1
+"""
+function stirling(x::Float64)
+  
+    a = zeros(18) 
+    if x < floatmin(Float64)*1000.0
+        return floatmax(Float64)/1000.0
+    elseif x < 1
+        return loggamma1p(x) - (x+0.5)*log(x) + x - log((2*pi))/2.0
+    elseif x < 2
+        return loggamma1p(x-1) - (x-0.5)*log(x) + x - log((2*pi))/2.0
+    elseif x < 3
+        return loggamma1p(x-2) - (x-0.5)*log(x) + x  - log((2*pi))/2.0 + log(x-1)
+    elseif x < 12
+        a[1]=1.996379051590076518221
```

Comment:
since `a` is constant, it could be assigned to a `const` variable outside the function (that way we won't have to allocate it). Maybe give it a more descriptive name though (say `stirling_chebyshev`).